sunpy/spectra/spectrum.py

```diff
@@ -37,6 +37,8 @@ def __new__(cls, data, *args, **kwargs):
 
     def __init__(self, data, freq_axis):
         self.freq_axis = freq_axis
+        if np.shape(data)[0] != np.shape(freq_axis)[0]:
```

Comment:
As written, the test should appear before the self.freq_axis = freq_axis assignment, since you want to test the input before adding properties to the object.